File: Sorting_Techniques/Selection_Sort/selection_sort.cpp
#include <bits/stdc++.h> 
using namespace std; 

void selectionSort(int arr[], int n) 
{ 
	int i, j, min_i; 
	for (i = 0; i < n-1; i++) 
	{ 
		min_i = i; 
		for (j = i+1; j < n; j++) 
		if (arr[j] < arr[min_i]) 
			min_i = j; 

        int temp=arr[min_i];
        arr[min_i]=arr[i];
        arr[i]=temp;
	} 
} 

int main() 
{ 
    int n;
    cin>>n;
    int arr[n];
    for(int i=0; i<n; i++)  cin>>arr[i];
	selectionSort(arr, n); 
	cout << "Sorted array: \n"; 
	for (int i=0; i < n; i++) 
		cout << arr[i] << " ";  
	return 0; 
}

File: Sorting_Techniques/Selection_Sort/selectionSort.js
function selectionSort(items) {
	var length = items.length;
	for (var i = 0; i < length - 1; i++) {
		var min = i; 
		for (var j = i + 1; j < length; j++) { 
			if (items[j] < items[min]) { 
				min = j; 
			}
		}
		if (min != i) {
			var tmp = items[i];
			items[i] = items[min];
			items[min] = tmp;
        }
    }
    return items;
}

var sorted = selectionSort([34,5,6,2,99,5,4,2,56,1]);
console.log(sorted);